src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -401,7 +401,13 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
             OPENVINO_ASSERT(weight_rank <= 3, "Currently only weights with equal to or less than 3D is supported");
             auto shift_size = std::max<size_t>(prim->input_size - 2, 0);
             int per_oc = PER_OC << shift_size;
-            int grouped = GROUPED | (1 << (prim->input_size - 1));
+            int grouped = 0;
+            if (weight_rank == 3) {
+                grouped = PER_TENSOR << (shift_size - 1);
+            } else {
+                grouped = GROUPED << shift_size;
+            }
```

Comment:
1. please update below too. https://github.com/openvinotoolkit/openvino/blob/b8f5b6f4ffa783d8d3e7fe60f61ae75fca78b554/src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp#L306 And could you please consolidate logic? Probably someone may make partial modifications in the future too. So please make it a single one.
2. I just curious, what part of this PR are you trying to fix (weight rank 2 or 3)? I'm not sure but it seems no changes compared before. Let's discuss offline.